core/common/src/main/java/alluxio/heartbeat/ScheduledTimer.java

```diff
@@ -69,7 +72,8 @@ protected void schedule() {
     try (LockResource r = new LockResource(mLock)) {
       Preconditions.checkState(!mScheduled, "Called schedule twice without waiting for any ticks");
       mScheduled = true;
-      mCondition.signal();
+      mTickCondition.signal();
+      HeartbeatScheduler.removeTimer(this);
```

Comment:
is it possible that the `signal` throws an exception, and `removeTimer` is not called? Would violate the desired invariant?